src/Model/Contact.php

```diff
@@ -1009,4 +1011,283 @@ public static function updateAvatar($avatar, $uid, $cid, $force = false)
 
 		return $data;
 	}
+
+	/**
+	 * @param integer $id contact id
+	 * @return boolean
+	 */
+	public static function update($id)
+	{
+		/*
+		Warning: Never ever fetch the public key via Probe::uri and write it into the contacts.
+		This will reliably kill your communication with Friendica contacts.
+		*/
+
+		$r = dba::select('contact', ['url', 'nurl', 'addr', 'alias', 'batch', 'notify', 'poll', 'poco', 'network'], ['id' => $id], ['limit' => 1]);
+		if (!DBM::is_result($r)) {
+			return false;
+		}
+
+		$ret = Probe::uri($r["url"]);
+
+		// If Probe::uri fails the network code will be different
+		if ($ret["network"] != $r["network"]) {
+			return false;
+		}
+
+		$update = false;
+
+		// make sure to not overwrite existing values with blank entries
+		foreach ($ret as $key => $val) {
+			if (isset($r[$key]) && ($r[$key] != "") && ($val == ""))
+				$ret[$key] = $r[$key];
+
+			if (isset($r[$key]) && ($ret[$key] != $r[$key]))
+				$update = true;
+		}
+
+		if (!$update) {
+			return true;
+		}
+
+		dba::update(
+			'contact',
+			[
+				'url' => $ret['url'],
+				'nurl' => normalise_link($ret['url']),
+				'addr' => $ret['addr'],
+				'alias' => $ret['alias'],
+				'batch' => $ret['batch'],
+				'notify' => $ret['notify'],
+				'poll' => $ret['poll'],
+				'poco' => $ret['poco']
+			],
+			['id' => $id]
+		);
+
+		// Update the corresponding gcontact entry
+		PortableContact::lastUpdated($ret["url"]);
+
+		return true;
+	}
+
+	/**
+	 * Takes a $uid and a url/handle and adds a new contact
+	 * Currently if the contact is DFRN, interactive needs to be true, to redirect to the
+	 * dfrn_request page.
+	 *
+	 * Otherwise this can be used to bulk add statusnet contacts, twitter contacts, etc.
+	 *
+	 * Returns an array
+	 * $return['success'] boolean true if successful
+	 * $return['message'] error text if success is false.
+	 *
+	 * @brief Takes a $uid and a url/handle and adds a new contact
+	 * @param int    $uid
+	 * @param string $url
+	 * @param bool   $interactive
+	 * @param string $network
+	 * @return boolean|string
+	 */
+	public static function add($uid, $url, $interactive = false, $network = '')
```

Comment:
Where `Contact::new` you renamed `new_contact` to?

Please rename it `Contact::create` instead.

Reply:
yes, renamed new_contact to add. There was no new() function. Am I understanding correctly?

regardless, i will rename add to create and update calls.

Reply:
The function calls currently are `Contact::new()` but the function you created is `Contact::add()`.

Reply:
Wow.. I didn't even see that!! You're right I'm amazed there were no errors, and I successfully connected with a new Friendica contact